packages/xo-server/src/xo-mixins/remotes.js

```diff
@@ -150,6 +151,16 @@ export default class {
     return remote.properties
   }
 
+  async assertEnabledRemote(remote) {
+    if (typeof remote === 'string') {
+      remote = await this._getRemote(remote)
+    }
+
+    if (!remote.enabled) {
+      throw new Error('remote is disabled')
+    }
+  }
```

Comment:
Would it make sense to merge this with `getRemoteWithCredentials`?

Reply:
I don't think so, because as its name says, it only returns the remote with its credentials.

Reply:
Yes, but it could do so only if the remote is enabled because it does not make sense to fetch the credentials of the remote if we are not allowed to use it, like it's done in `getRemoteHandler`.

It's only an suggestion, the idea behind it is that I want to ensure that we will never forget to call `assertEnabledRemote` and use a remote if it's disabled.

Reply:
Make sense :+1: